frame/contracts/src/exec.rs

```diff
@@ -1016,6 +1020,19 @@ where
 		)
 	}
 
+	fn set_code_hash(&mut self, hash: CodeHash<Self::T>) -> Result<(), DispatchError> {
+		let top_frame = &mut self.top_frame_mut();
```

Comment:
It is already a ref.

```suggestion
		let top_frame = self.top_frame_mut();
```